StateUtil.verifySnapshotLogConsistent message does not indicate which partition is affected

**Description**

In case on a snapshot inconsistency the broker fails to startup and logs the IllegalArgumentException that contains the information on the index mismatch of the log and the snapshot, it would be helpful to also include the partition id.

```
[Broker-7-Startup] [Broker-7-zb-actors-2] WARN 
        io.camunda.zeebe.broker.system - Aborting startup process due to exception during step Partition Manager
  java.util.concurrent.CompletionException: java.lang.IllegalStateException: Expected to find a snapshot at index >= log's first index 161591, but found snapshot 25. A previous snapshot is most likely corrupted.
```

<details>
  <summary>Logs</summary>

  ```
  2023-05-21 14:44:55.018 [Broker-7-Startup] [Broker-7-zb-actors-2] WARN 
        io.camunda.zeebe.broker.system - Aborting startup process due to exception during step Partition Manager
  java.util.concurrent.CompletionException: java.lang.IllegalStateException: Expected to find a snapshot at index >= log's first index 161591, but found snapshot 25. A previous snapshot is most likely corrupted.
	  at java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]
	  at java.util.concurrent.CompletableFuture.uniApplyNow(Unknown Source) ~[?:?]
	  at java.util.concurrent.CompletableFuture.uniApplyStage(Unknown Source) ~[?:?]
	  at java.util.concurrent.CompletableFuture.thenApply(Unknown Source) ~[?:?]
	  at io.atomix.raft.partition.RaftPartition.open(RaftPartition.java:90) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
	  at io.atomix.raft.partition.RaftPartitionGroup.lambda$join$2(RaftPartitionGroup.java:174) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
	  at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	  at java.util.HashMap$KeySpliterator.forEachRemaining(Unknown Source) ~[?:?]
	  at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	  at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	  at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	  at java.util.stream.AbstractPipeline.evaluateToArrayNode(Unknown Source) ~[?:?]
	  at java.util.stream.ReferencePipeline.toArray(Unknown Source) ~[?:?]
	  at io.atomix.raft.partition.RaftPartitionGroup.join(RaftPartitionGroup.java:175) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
	  at io.atomix.primitive.partition.impl.DefaultPartitionService.start(DefaultPartitionService.java:63) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.broker.partitioning.PartitionManagerImpl.start(PartitionManagerImpl.java:125) ~[zeebe-broker-8.1.9.jar:8.1.9]
	  at java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source) ~[?:?]
	  at java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source) ~[?:?]
	  at java.util.concurrent.ForkJoinTask.doExec(Unknown Source) ~[?:?]
	  at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source) ~[?:?]
	  at java.util.concurrent.ForkJoinPool.scan(Unknown Source) ~[?:?]
	  at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source) ~[?:?]
	  at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) ~[?:?]
  Caused by: java.lang.IllegalStateException: Expected to find a snapshot at index >= log's first index 161591, but found snapshot 25. A previous snapshot is most likely corrupted.
	  at io.atomix.raft.utils.StateUtil.verifySnapshotLogConsistent(StateUtil.java:32) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
	  at io.atomix.raft.impl.RaftContext.<init>(RaftContext.java:204) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
	  at io.atomix.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:263) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
	  at io.atomix.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:237) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
	  at io.atomix.raft.partition.impl.RaftPartitionServer.buildServer(RaftPartitionServer.java:185) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
	  at io.atomix.raft.partition.impl.RaftPartitionServer.initServer(RaftPartitionServer.java:151) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
	  at io.atomix.raft.partition.impl.RaftPartitionServer.start(RaftPartitionServer.java:110) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
	  ... 19 more
  2023-05-21 14:44:55.025 [] [main] ERROR
        io.camunda.zeebe.broker.system - Failed to start broker 7!
  java.util.concurrent.ExecutionException: Startup failed in the following steps: [Partition Manager]. See suppressed exceptions for details.
	  at io.camunda.zeebe.scheduler.future.CompletableActorFuture.get(CompletableActorFuture.java:142) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.future.CompletableActorFuture.get(CompletableActorFuture.java:109) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.FutureUtil.join(FutureUtil.java:21) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.future.CompletableActorFuture.join(CompletableActorFuture.java:198) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.broker.Broker.internalStart(Broker.java:101) ~[zeebe-broker-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.util.LogUtil.doWithMDC(LogUtil.java:23) ~[zeebe-util-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.broker.Broker.start(Broker.java:83) ~[zeebe-broker-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.broker.StandaloneBroker.run(StandaloneBroker.java:92) ~[camunda-zeebe-8.1.9.jar:8.1.9]
	  at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:771) ~[spring-boot-2.7.4.jar:2.7.4]
	  at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:755) ~[spring-boot-2.7.4.jar:2.7.4]
	  at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) ~[spring-boot-2.7.4.jar:2.7.4]
	  at io.camunda.zeebe.broker.StandaloneBroker.main(StandaloneBroker.java:82) ~[camunda-zeebe-8.1.9.jar:8.1.9]
  Caused by: io.camunda.zeebe.scheduler.startup.StartupProcessException: Startup failed in the following steps: [Partition Manager]. See suppressed exceptions for details.
	  at io.camunda.zeebe.scheduler.startup.StartupProcess.aggregateExceptionsSynchronized(StartupProcess.java:282) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.startup.StartupProcess.completeStartupFutureExceptionallySynchronized(StartupProcess.java:183) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.startup.StartupProcess.lambda$proceedWithStartupSynchronized$3(StartupProcess.java:167) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:33) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:94) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:45) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:119) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:106) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:87) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:198) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  Suppressed: io.camunda.zeebe.scheduler.startup.StartupProcessStepException: Bootstrap step Partition Manager failed
		  at io.camunda.zeebe.scheduler.startup.StartupProcess.completeStartupFutureExceptionallySynchronized(StartupProcess.java:185) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
		  at io.camunda.zeebe.scheduler.startup.StartupProcess.lambda$proceedWithStartupSynchronized$3(StartupProcess.java:167) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
		  at io.camunda.zeebe.scheduler.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:33) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
		  at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:94) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
		  at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:45) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
		  at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:119) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
		  at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:106) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
		  at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:87) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
		  at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:198) ~[zeebe-scheduler-8.1.9.jar:8.1.9]
	  Caused by: java.util.concurrent.CompletionException: java.lang.IllegalStateException: Expected to find a snapshot at index >= log's first index 161591, but found snapshot 25. A previous snapshot is most likely corrupted.
		  at java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]
		  at java.util.concurrent.CompletableFuture.uniApplyNow(Unknown Source) ~[?:?]
		  at java.util.concurrent.CompletableFuture.uniApplyStage(Unknown Source) ~[?:?]
		  at java.util.concurrent.CompletableFuture.thenApply(Unknown Source) ~[?:?]
		  at io.atomix.raft.partition.RaftPartition.open(RaftPartition.java:90) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.atomix.raft.partition.RaftPartitionGroup.lambda$join$2(RaftPartitionGroup.java:174) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
		  at java.util.HashMap$KeySpliterator.forEachRemaining(Unknown Source) ~[?:?]
		  at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
		  at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
		  at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
		  at java.util.stream.AbstractPipeline.evaluateToArrayNode(Unknown Source) ~[?:?]
		  at java.util.stream.ReferencePipeline.toArray(Unknown Source) ~[?:?]
		  at io.atomix.raft.partition.RaftPartitionGroup.join(RaftPartitionGroup.java:175) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.atomix.primitive.partition.impl.DefaultPartitionService.start(DefaultPartitionService.java:63) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.camunda.zeebe.broker.partitioning.PartitionManagerImpl.start(PartitionManagerImpl.java:125) ~[zeebe-broker-8.1.9.jar:8.1.9]
		  at java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source) ~[?:?]
		  at java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source) ~[?:?]
		  at java.util.concurrent.ForkJoinTask.doExec(Unknown Source) ~[?:?]
		  at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source) ~[?:?]
		  at java.util.concurrent.ForkJoinPool.scan(Unknown Source) ~[?:?]
		  at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source) ~[?:?]
		  at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) ~[?:?]
	  Caused by: java.lang.IllegalStateException: Expected to find a snapshot at index >= log's first index 161591, but found snapshot 25. A previous snapshot is most likely corrupted.
		  at io.atomix.raft.utils.StateUtil.verifySnapshotLogConsistent(StateUtil.java:32) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.atomix.raft.impl.RaftContext.<init>(RaftContext.java:204) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.atomix.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:263) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.atomix.raft.impl.DefaultRaftServer$Builder.build(DefaultRaftServer.java:237) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.atomix.raft.partition.impl.RaftPartitionServer.buildServer(RaftPartitionServer.java:185) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.atomix.raft.partition.impl.RaftPartitionServer.initServer(RaftPartitionServer.java:151) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.atomix.raft.partition.impl.RaftPartitionServer.start(RaftPartitionServer.java:110) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.atomix.raft.partition.RaftPartition.open(RaftPartition.java:90) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.atomix.raft.partition.RaftPartitionGroup.lambda$join$2(RaftPartitionGroup.java:174) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
		  at java.util.HashMap$KeySpliterator.forEachRemaining(Unknown Source) ~[?:?]
		  at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
		  at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
		  at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
		  at java.util.stream.AbstractPipeline.evaluateToArrayNode(Unknown Source) ~[?:?]
		  at java.util.stream.ReferencePipeline.toArray(Unknown Source) ~[?:?]
		  at io.atomix.raft.partition.RaftPartitionGroup.join(RaftPartitionGroup.java:175) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.atomix.primitive.partition.impl.DefaultPartitionService.start(DefaultPartitionService.java:63) ~[zeebe-atomix-cluster-8.1.9.jar:8.1.9]
		  at io.camunda.zeebe.broker.partitioning.PartitionManagerImpl.start(PartitionManagerImpl.java:125) ~[zeebe-broker-8.1.9.jar:8.1.9]
		  at java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source) ~[?:?]
		  at java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source) ~[?:?]
		  at java.util.concurrent.ForkJoinTask.doExec(Unknown Source) ~[?:?]
		  at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source) ~[?:?]
		  at java.util.concurrent.ForkJoinPool.scan(Unknown Source) ~[?:?]
		  at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source) ~[?:?]
		  at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) ~[?:?]
  ```
</details>

--- a/atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java
+++ b/atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java
@@ -208,7 +208,12 @@ public class RaftContext implements AutoCloseable, HealthMonitorable {
         .getLatestSnapshot()
         .ifPresent(persistedSnapshot -> currentSnapshot = persistedSnapshot);
     StateUtil.verifySnapshotLogConsistent(
-        getCurrentSnapshotIndex(), raftLog.getFirstIndex(), raftLog.isEmpty(), raftLog::reset, log);
+        partitionId,
+        getCurrentSnapshotIndex(),
+        raftLog.getFirstIndex(),
+        raftLog.isEmpty(),
+        raftLog::reset,
+        log);
 
     logCompactor =
         new LogCompactor(

--- a/atomix/cluster/src/main/java/io/atomix/raft/utils/StateUtil.java
+++ b/atomix/cluster/src/main/java/io/atomix/raft/utils/StateUtil.java
@@ -14,6 +14,7 @@ public class StateUtil {
 
   /** throws IllegalStateException if state is inconsistent */
   public static void verifySnapshotLogConsistent(
+      final long partitionId,
       final long snapshotIndex,
       final long firstIndex,
       final boolean isLogEmpty,
@@ -30,15 +31,15 @@ public class StateUtil {
       // There is a gap between snapshot and first log entry
       throw new IllegalStateException(
           String.format(
-              "Expected to find a snapshot at index >= log's first index %d, but found snapshot %d. A previous snapshot is most likely corrupted.",
-              firstIndex, snapshotIndex));
+              "In partition %d expected to find a snapshot at index >= log's first index %d,"
+                  + " but found snapshot %d. A previous snapshot is most likely corrupted.",
+              partitionId, firstIndex, snapshotIndex));
     } else {
       log.info(
-          "Current snapshot index ({}) is lower than log's first index {}. But the log is empty. Most likely the node crashed while committing a snapshot at index {}. Resetting log to {}",
-          snapshotIndex,
-          firstIndex,
-          firstIndex - 1,
-          snapshotIndex + 1);
+          "In partition %d current snapshot index ({}) is lower than log's first index {}. "
+              + "But the log is empty. Most likely the node crashed while committing a snapshot "
+              + "at index {}. Resetting log to {}",
+          partitionId, snapshotIndex, firstIndex, firstIndex - 1, snapshotIndex + 1);
       logResetter.accept(snapshotIndex + 1);
     }
   }

--- a/atomix/cluster/src/test/java/io/atomix/raft/utils/StateUtilTest.java
+++ b/atomix/cluster/src/test/java/io/atomix/raft/utils/StateUtilTest.java
@@ -32,7 +32,12 @@ class StateUtilTest {
         .isThrownBy(
             () ->
                 StateUtil.verifySnapshotLogConsistent(
-                    state.snapshotIndex(), state.firstIndex(), state.isLogEmpty(), i -> {}, LOG));
+                    1L,
+                    state.snapshotIndex(),
+                    state.firstIndex(),
+                    state.isLogEmpty(),
+                    i -> {},
+                    LOG));
   }
 
   @ParameterizedTest
@@ -42,6 +47,7 @@ class StateUtilTest {
         .isThrownBy(
             () ->
                 StateUtil.verifySnapshotLogConsistent(
+                    1L,
                     state.snapshotIndex(),
                     state.firstIndex(),
                     state.isLogEmpty(),
@@ -59,6 +65,7 @@ class StateUtilTest {
         .isThrownBy(
             () ->
                 StateUtil.verifySnapshotLogConsistent(
+                    1L,
                     state.snapshotIndex(),
                     state.firstIndex(),
                     state.isLogEmpty(),

